plugins/objects/base-things/src/main/java/com/freedomotic/things/impl/Fridge.java

```diff
@@ -0,0 +1,127 @@
+import com.freedomotic.behaviors.RangedIntBehaviorLogic;
+import com.freedomotic.reactions.Command;
+import com.freedomotic.reactions.CommandPersistence;
+
+/**
+ *
+ * @author Mauro Cicolella
+ */
+public class Fridge
+        extends ElectricDevice {
+
+    private RangedIntBehaviorLogic fridgeTemperature;
+    private RangedIntBehaviorLogic freezerTemperature;
+    protected final static String BEHAVIOR_FRIDGE_TEMPERATURE = "fridge-temperature";
+    protected final static String BEHAVIOR_FREEZER_TEMPERATURE = "freezer-temperature";
+
+    @Override
+    public void init() {
+        //linking this property with the behavior defined in the XML
+        fridgeTemperature = new RangedIntBehaviorLogic((RangedIntBehavior) getPojo().getBehavior(BEHAVIOR_FRIDGE_TEMPERATURE));
+        freezerTemperature = new RangedIntBehaviorLogic((RangedIntBehavior) getPojo().getBehavior(BEHAVIOR_FREEZER_TEMPERATURE));
+
+        fridgeTemperature.addListener(new RangedIntBehaviorLogic.Listener() {
+
+            @Override
+            public void onLowerBoundValue(Config params, boolean fireCommand) {
+            }
+
+            @Override
+            public void onUpperBoundValue(Config params, boolean fireCommand) {
+            }
+
+            @Override
+            public void onRangeValue(int rangeValue, Config params, boolean fireCommand) {
+                executeSetFridgeTemperature(rangeValue, params);
+            }
+        });
+
+        freezerTemperature.addListener(new RangedIntBehaviorLogic.Listener() {
+
+            @Override
+            public void onLowerBoundValue(Config params, boolean fireCommand) {
+            }
+
+            @Override
+            public void onUpperBoundValue(Config params, boolean fireCommand) {
+            }
+
+            @Override
+            public void onRangeValue(int rangeValue, Config params, boolean fireCommand) {
+                executeSetFreezerTemperature(rangeValue, params);
+            }
+        });
+
+        //register new behaviors to the superclass to make it visible to it
+        registerBehavior(fridgeTemperature);
+        registerBehavior(freezerTemperature);
+        super.init();
+    }
+
+    @Override
+    public void executePowerOff(Config params) {
+        /*
+         * executeCommand the body of the super implementation The super call
+         * must be the last call as it executes setChanged(true)
+         */
+        super.executePowerOff(params);
+    }
+
+    @Override
+    public void executePowerOn(Config params) {
+        //executeCommand the body of the super implementation
+        super.executePowerOn(params);
+    }
+
+    public void executeSetFridgeTemperature(int rangeValue, Config params) {
+        boolean executed = executeCommand("set fridge temperature", params); //executes the developer level command associated with 'set brightness' action
+
+        if (executed) {
+            fridgeTemperature.setValue(rangeValue);
```

Comment:
would be good to move it in a "setFridgeTemperature(int value)" method called inside this if. The same for the freezer temperature